makelove/macos.py

```diff
@@ -0,0 +1,245 @@
+            <string>icon.icns</string>
+            <key>CFBundleIdentifier</key>
+            <string>{metadata['BundleIdentifier']}</string>
+            <key>CFBundleInfoDictionaryVersion</key>
+            <string>6.0</string>
+            <key>CFBundleName</key>
+            <string>{metadata['ProductName']}</string>
+            <key>CFBundlePackageType</key>
+            <string>APPL</string>
+            <key>CFBundleShortVersionString</key>
+            <string>{metadata['FileVersion']}</string>
+            <key>CFBundleSignature</key>
+            <string>LoVe</string>
+            <key>CFBundleSupportedPlatforms</key>
+            <array>
+                <string>MacOSX</string>
+            </array>
+            <key>DTCompiler</key>
+            <string>com.apple.compilers.llvm.clang.1_0</string>
+            <key>DTPlatformBuild</key>
+            <string>11C504</string>
+            <key>DTPlatformVersion</key>
+            <string>GM</string>
+            <key>DTSDKBuild</key>
+            <string>19B90</string>
+            <key>DTSDKName</key>
+            <string>macosx10.15</string>
+            <key>DTXcode</key>
+            <string>1130</string>
+            <key>DTXcodeBuild</key>
+            <string>11C504</string>
+            <key>LSApplicationCategoryType</key>
+            <string>public.app-category.games</string>
+            <key>LSMinimumSystemVersion</key>
+            <string>10.7</string>
+            <key>NSHighResolutionCapable</key>
+            <true/>
+            <key>NSHumanReadableCopyright</key>
+            <string>{metadata['LegalCopyright']}</string>
+            <key>NSPrincipalClass</key>
+            <string>NSApplication</string>
+            <key>NSSupportsAutomaticGraphicsSwitching</key>
+            <false/>
+        </dict>
+        </plist>
+    """
+
+
+def build_macos(config, version, target, target_directory, love_file_path):
+    if target in config and "love_binaries" in config[target]:
+        love_binaries = config[target]["love_binaries"]
+    else:
+        assert "love_version" in config
+        print("No love binaries specified for target {}".format(target))
+        love_binaries = get_default_love_binary_dir(config["love_version"], target)
+        if os.path.isdir(love_binaries):
+            print("Love binaries already present in '{}'".format(love_binaries))
+        else:
+            download_love(config["love_version"], target)
+
+    src = os.path.join(love_binaries, 'love.zip')
+    dst = os.path.join(target_directory, f"{config['name']}-{target}.zip")
+    with open(src, 'rb') as lovef, ZipFile(lovef) as loveBinaryZip, \
+            open(dst, 'wb+') as outf, ZipFile(outf, mode='w') as appZip, \
+            open(love_file_path, 'rb') as loveZip:
+
+        for zipinfo in loveBinaryZip.infolist():
+            if not zipinfo.filename.startswith("love.app/"):
+                raise RuntimeError("Got bad or unxpexpectedly formatted love zip file")
```

Comment:
It's probably not a bad idea to use RuntimeError for this sort of thing, but everywhere else I use `sys.exit` (and you did in most places too). I think it should be the same everywhere.